logstash-core/lib/logstash/settings.rb

```diff
@@ -176,18 +176,43 @@ def validate(value)
       end
     end
 
-    ### Specific settings #####
-
-    class Boolean < Setting
-      def initialize(name, default, strict=true, &validator_proc)
+    class Coercible < Setting
```

Comment:
Yes, some actually try to coerce the input into its type, while most of the others don't.
For example the String setting will fail for something which isn't a String, a Numeric will try to convert a String into a Numeric